### ISSUE TITLE
Spatial discretization of heat equation isn't proper second order central derivative.

### ISSUE THREAD

**Issue by castellinilinguini** [Systems-Theory-in-Systems-Biology/EPI#78]
**Describe the bug**
The current implementation of the right-hand side of the heat equation looks like this:
```python
du_dx = jnp.gradient(u, dx, axis=0)
du_dy = jnp.gradient(u, dy, axis=1)
du_dx2 = jnp.gradient(du_dx, dx, axis=0)
du_dy2 = jnp.gradient(du_dy, dy, axis=1)
```
According to the documentation, jnp.gradient is equivalent to $\frac{f(x_{i+1}) - f(x_{i-1})}{2h}$, so the second order derivatives here are computed as $\frac{f(x_{i+2}) - 2f(x_i) + f(x_{i-2})}{4h^2}$ and not as standard second order central second derivatives $\frac{f(x_{i+1}) - 2f(x_i) + f(x_{i-1})}{h^2}$. I believe this might also cause some numerical issue, which in turn might cause some of the issues with bad fits for marginal EPI.
